testgrid/config.yaml

```diff
@@ -0,0 +1,5 @@
+# Dashboards need to be specified here to be created on TestGrid
+# A prow annotation will be invalid if it references a dashboard that doesn't exist
+dashboards:
```

Comment:
Do we need a) this file and b) these keys in it?

Reply:
We do need this file, as Configurator needs a base config.yaml file to append Prow Annotations to. Configurator doesn't create TestGrid YAML out of nothing.

As the comment says, dashboards will need to be enumerated here. A shell file is needed anyway, so why not add the correct keys and a description for what will go here?

Reply:
IMO if it works with this file and fails without it, then (long-term) we should remove that constraint because its pretty ridiculous. There's nothing meaningful in this file.

Long term, do we really need this file? Why can't I annotate my jobs with dashboard, dashboard group, tab options, etc and have it generate everything?

Reply:
The big advantage to having the list of dashboards explicitly like this is that Configurator can point out (and fail) when you misspell a dashboard. If there's no "source of truth" for that, then there's no way for Configurator to know whether `sig-tset-infra-king` is incorrect, or whether that's my brand new dashboard.

I think it's more valuable to have a singular, though separate, list of dashboards and have annotations tie them to them.

Reply:
That doesn't seem very convincing to me.

a) we can still validate that dashboards all start with sig- or google- like we do today
b) nothing preventing someone from fat fingering things in this yaml file
c) it is trivial to send a follow up PR to change sig-testing-king to sig-testing-kind

The test group and its thousands of GCS files are the only thing that's consequential to rename (aka if you rename the testgroup path you lose test history)